public/js/specberus.js

```diff
@@ -192,35 +199,62 @@ jQuery.extend({
         attachCustomScroll();
         // endValidation();
     });
+    socket.on("finishedExtraction", function (data) {
+      $(".manual").toggle(true);
+      var processDocument = (data.processdocument && data.processDocument.indexOf('Process-20051014') > -1) ? "2005" : "2015";
+      $profile.val(data.profile);
+      $processDocument.find("label#2005").toggleClass("active", (processDocument === "2005"));
+      $processDocument.find("label#2015").toggleClass("active", (processDocument === "2015"));
+      $noRecTrack.prop('checked', !data.rectrack);
+      $informativeOnly.prop('checked', data.informative);
+      $validation.val('simple-validation');
+      var options = {
+                        "url"             : data.url
+                      , "profile"         : data.profile
+                      , "validation"      : 'simple-validation'
+                      , "noRecTrack"      : !data.rectrack || false
+                      , "informativeOnly" : data.informative || false
+                      , "echidnaReady"    : false
+                      , "patentPolicy"    : "pp2004"
+                      , "processDocument" : processDocument
+                    };
+      validate(options);
+      var newurl = document.URL.split('?')[0] + "?" + $.param(options)
+      history.pushState(options, url + " - " + profile, newurl);
+    });
 
     // handle the form
     $("#options").submit(function () {
         clearError();
-        var url = $url.val()
-        ,   profile = $profile.val()
-        ,   validation = $validation.val()
-        ,   noRecTrack = $noRecTrack.is(":checked") || false
-        ,   informativeOnly = $informativeOnly.is(":checked") || false
-        ,   echidnaReady = $echidnaReady.is(":checked") || false
-        ,   patentPolicy = $patentPolicy.find('label.active').attr('id')
-        ,   processDocument = $processDocument.find('label.active').attr('id')
-        ;
-        if (!url) showError("Missing URL parameter.");
-        if (!profile) showError("Missing profile parameter.");
-        if (echidnaReady) profile += '-Echidna';
```

Comment:
I think echidna is a very special case of Specberus. When you select that option, you can't expect specberus to check a FPWD, CR, PR or REC.
Furthermore, when the auto profile is selected, Specberus will ignore the options (informative, rec-track, profile, etc) so it can be confusing if the option echidna is treated differently.